Extension/src/LanguageServer/settings.ts

```diff
@@ -312,88 +309,199 @@ export class CppSettings extends Settings {
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
+    }
+
+    private getAsNumber(settingName: string): number {
+        const value: any | undefined | null = super.Section.get(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+
+        if (isNumber(value)) {
+            if (setting.minimum !== undefined && setting.maximum !== undefined) { 
+                if (value >= setting.minimum && value <= setting.maximum) { // Check if the value is within the valid range.
+                    return value;
+                }
+            } else {
+                return value;
+            }
+        }   
+
+        return setting.default;
+    }
+
+    // This helper function returns the value of a setting as an array of strings with proper type validation.
+    // Additionally, it checks for valid enum values if an array of enums is detected.
+    private getAsArrayOfStrings(settingName: string, allowUndefinedEnums: boolean = false): string[] {
+        const setting = getRawSetting("C_Cpp." + settingName);
+        const value: any | undefined | null = super.Section.get(settingName);
+        if (isArrayOfString(value)) {
+            if (setting.items.enum !== undefined) {
+                if (!value.every(x => this.validateEnum(setting.items.enum, x))) {
+                    if (allowUndefinedEnums) {
+                        return value
+                    } else {
+                        return setting.default;
+                    }
+                } else {
+                    return value;
+                }
+            }
+            return value;
+        }
+        return setting.default;
+    }
+
+    // This helper function returns the value of a setting as a key-value object with proper type validation.
+    private getAsKeyValueObject(settingName: string, keyType: string, valueType: string): any {
+        const value: any = super.Section.get(settingName);
+        if (this.isValidMapping(value, keyType, valueType)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
+    }
+
+    // This helper function validates whether the given value is a valid enum value for the given setting.
+    private isValidEnum(settingName: string, value: any): boolean {
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (this.validateEnum(setting.enum, value)) {
+            return value;
+        }
+        return setting.default;
+    }
+
+    // This helper function validates whether the given object is a valid mapping of key and value type.
+    // EX: {"key": true, "key2": false} should return true for keyType = string and valueType = boolean.
+    private isValidMapping(value: any, keyType: string, valueType: string): boolean {
+        if (value === undefined || value === null) { return false; }
+        for (const key of Object.keys(value)) {
```

Comment:
Take a look at `Object.entries` as a way to get the key and value at the same time instead of needing to do `value[key]` later.

Reply:
because then you could also use the `.every()` function on the array that is returned instead of writing a for loop here.